components/newsletter-signup/SignupForm.jsx

```diff
@@ -42,25 +35,28 @@ var SignupForm = React.createClass({
         email: this.state.email,
         trigger_welcome: 'N'
       })
-      .end(function(err, res){
-        if ( err ) { 
-          // TODO: add error handling
-          return; 
-        }
-
-        if ( res.statusCode === 200 ) {
-          window.location = config.ORIGIN+"?signup=thanks";
+      .end(function(err, res) {
+        if (res.statusCode >= 200 && res.statusCode < 300) {
+          self.setState({validationErrorType: null}, function() {
+            window.location = config.ORIGIN+"?signup=thanks";
+          });
         } else {
-          // TODO: add error handling
+          // for basket error code references, 
+          // see https://github.com/mozilla/basket-client/blob/master/basket/errors.py
+          var basketErrorType = JSON.parse(res.text).code == 2 ? "email" : "other";
+          self.setState( {validationErrorType: basketErrorType} );
         }
       });
 
   },
   renderValidationErrors: function() {
-    if (this.state.validationErrors.length) {
+    var errorType = this.state.validationErrorType;
+    if ( errorType ) {
+      var errorMsg = errorType === "email" ? "Please enter a valid email address." : "There's been a problem with our system. Please try again later.";
```

Comment:
cool, I thought about it when I wrote this part of code. Guess I made the wrong choice 😉 Will fix now.